src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,50 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int count = 0;
+        int[] mainDiagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            for (int j = 0; j < matrix[i].length; j++) {
+                if (i == j) {
+                    mainDiagonal[count] = matrix[i][j];
+                    count++;
+                }
+            }
+        }
+        return mainDiagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] secondDiagonal = new int[matrix.length];
+        int n = matrix.length;
```

Comment:
let's create more understandable name of the variable instead of `n`

Reply:
No problem, what about length?